app/internal/page/certificateprovider/enter_mobile_number.go

```diff
@@ -0,0 +1,74 @@
+package certificateprovider
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/page"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/validation"
+)
+
+type mobileNumberData struct {
+	App    page.AppData
+	Lpa    *page.Lpa
+	Form   *mobileNumberForm
+	Errors validation.List
+}
+
+type mobileNumberForm struct {
+	Mobile string
+}
+
+func EnterMobileNumber(tmpl template.Template, lpaStore LpaStore) page.Handler {
+	return func(appData page.AppData, w http.ResponseWriter, r *http.Request) error {
+		lpa, err := lpaStore.Get(r.Context())
+		if err != nil {
+			return err
+		}
+
+		data := &mobileNumberData{
+			App: appData,
+			Lpa: lpa,
+			Form: &mobileNumberForm{
+				Mobile: lpa.CertificateProviderProvidedDetails.Mobile,
+			},
+		}
+
+		if r.Method == http.MethodPost {
+			data.Form = readMobileNumberForm(r)
+			data.Errors = data.Form.Validate()
+
+			if data.Errors.None() {
+				lpa.CertificateProviderProvidedDetails.Mobile = data.Form.Mobile
+
+				if err := lpaStore.Put(r.Context(), lpa); err != nil {
+					return err
+				}
+
+				return appData.Redirect(w, r, lpa, page.Paths.CertificateProviderYourAddress)
+			}
+		}
+
+		return tmpl(w, data)
+	}
+}
+
+func readMobileNumberForm(r *http.Request) *mobileNumberForm {
+	return &mobileNumberForm{
+		Mobile: page.PostFormString(r, "mobile"),
+	}
+}
+
+func (f *mobileNumberForm) Validate() validation.List {
+	var errors validation.List
+
+	errors.String("mobile", "yourUkMobile", strings.ReplaceAll(f.Mobile, " ", ""),
+		validation.Empty())
+
+	if !validation.MobileRegex.MatchString(f.Mobile) {
```

Comment:
can we just use the existing validation for now, and come back to these after the error chat?

Reply:
or if we really, really must use a different error to the label, add a `MobileCheck.Label(string) MobileCheck` method to customise it

Reply:
Fair point, I'll go with defaults for now 👍